homeassistant/components/assist_satellite/__init__.py

```diff
@@ -0,0 +1,63 @@
+"""Base class for assist satellite entities."""
+
+import logging
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import config_validation as cv, entity_registry as er
+from homeassistant.helpers.entity_component import EntityComponent
+from homeassistant.helpers.typing import ConfigType
+
+from .const import DOMAIN
+from .entity import AssistSatelliteEntity
+from .models import AssistSatelliteEntityFeature, AssistSatelliteState, SatelliteConfig
+
+__all__ = [
+    "DOMAIN",
+    "AssistSatelliteEntityFeature",
+    "AssistSatelliteState",
+    "AssistSatelliteEntity",
+    "SatelliteConfig",
+]
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORM_SCHEMA = cv.PLATFORM_SCHEMA
+PLATFORM_SCHEMA_BASE = cv.PLATFORM_SCHEMA_BASE
+
+
+async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
+    component = hass.data[DOMAIN] = EntityComponent[AssistSatelliteEntity](
+        _LOGGER, DOMAIN, hass
+    )
+    await component.async_setup(config)
+
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up a config entry."""
+    component: EntityComponent[AssistSatelliteEntity] = hass.data[DOMAIN]
+    return await component.async_setup_entry(entry)
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Unload a config entry."""
+    component: EntityComponent[AssistSatelliteEntity] = hass.data[DOMAIN]
+    return await component.async_unload_entry(entry)
+
+
+def async_get_satellite_entity(
```

Comment:
This is only used in tests, move this out of prod code.

Reply:
I'm using this in the `make_protocol` function in `voip.py`

Reply:
Can we have entity register its start stream function with VoipDevice inside async added to hass ? We don't want to expose raw entity objects to other code.

Reply:
Added a `protocol` field to the VoIP device that's used now in `make_protocol` instead of getting the entity.